### PR TITLE
[PVR] Refactor: Encapsulate recording path

### PR DESCRIPTION
**Pull request by ksooo**
No functional changes, only refactoring. Put all recordings path string manipulation magic into one specialized class.

@Jalle19 @xhaggi mind taking a look
@Razzeee maybe you can supply the needed VS project files update?

### CODE REVIEW

File: xbmc/pvr/recordings/PVRRecordingsPath.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                       const std::string &strDirectory, const std::string &strTitle,
+                       int iSeason, int iEpisode, int iYear,
+                       const std::string &strSubtitle, const std::string &strChannelName,
+                       const CDateTime &recordingTime);
+
+    operator std::string() const { return m_path; }
+
+    bool IsValid() const { return m_bValid; }
+
+    const std::string &GetPath() const         { return m_path; }
```
Comment (FernetMenta) [inline xbmc/pvr/recordings/PVRRecordingsPath.h]: why do you do this imo silly vertical alignment? sooner or later someone adds a member that does not fit and needs to adjust the entire block. as a result the diff is no usable.

Comment (ksooo) [inline xbmc/pvr/recordings/PVRRecordingsPath.h]: will correct this "imo silly" alignment. we do not have any coding guidelines for this, but if it helps...

**Comment by Razzeee** [comment]
@ksooo pick https://github.com/Razzeee/xbmc/commit/a3fea65b908ca1d5916ae054d25e2b2086b399af

**Comment by ksooo** [comment]
thx @Razzeee

**Comment by xhaggi** [comment]
looks good after a quick look.

**Comment by ksooo** [comment]
jenkins build this please

**Comment by Jalle19** [comment]
Good improvement, thanks a lot!

**Comment by ksooo** [comment]
Rebased and <code>CPVRRecordingsPath::Init</code> now returning <code>void</code>.

**Comment by ksooo** [comment]
jenkins build this please

File: xbmc/video/windows/GUIWindowVideoBase.cpp [hunk 2]
```diff
@@ -1125,7 +1126,9 @@ bool CGUIWindowVideoBase::OnPlayMedia(int iItem, const std::string &player)
   }
   CLog::Log(LOGDEBUG, "%s %s", __FUNCTION__, CURL::GetRedacted(item.GetPath()).c_str());
 
-  if (StringUtils::StartsWith(item.GetPath(), "pvr://recordings/active/"))
+
+  CPVRRecordingsPath path(item.GetPath());
```
Comment (FernetMenta) [inline xbmc/video/windows/GUIWindowVideoBase.cpp]: this causes issues (identifies other files as recordings). don't touch any item that does not start with pvr://
this old code was better in this case

Comment (FernetMenta) [inline xbmc/video/windows/GUIWindowVideoBase.cpp]: Note that it does not make much sense to construct a PVR related object every time a video is started. The majority of used does not even use pvr. This is a waste of resources.

**Comment by ksooo** [comment]
I agree on the second comment regarding waste if resources.

The false positives created by the change mentioned advice is a stupid bug in the impression of CPVRRecordingsPath::IsValid() which I will fix. Thanks for pointing this out.

**Pull request merged**